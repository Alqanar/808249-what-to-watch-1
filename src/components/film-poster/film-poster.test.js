import React from "react";
import renderer from "react-test-renderer";

import FilmPoster from "./film-poster.tsx";


describe(`Poster of the film correctly renders`, () => {
  it(`when isBig not passed div render with class 'movie-card__poster'`, () => {
    const tree = renderer
      .create(<FilmPoster
        name={`Avengers: Endgame`}
        posterLink={`https://st.kp.yandex.net/images/film_iphone/iphone360_843650.jpg`}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`when size L div render with classes 'movie-card__poster' and 'movie-card__poster--big'`, () => {
    const tree = renderer
      .create(<FilmPoster
        name={`Avengers: Endgame`}
        posterLink={`https://st.kp.yandex.net/images/film_iphone/iphone360_843650.jpg`}
        size="L"
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`when size S div render with classes 'movie-card__poster' and 'movie-card__poster--small'`, () => {
    const tree = renderer
      .create(<FilmPoster
        name={`Avengers: Endgame`}
        posterLink={`https://st.kp.yandex.net/images/film_iphone/iphone360_843650.jpg`}
        size="S"
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
