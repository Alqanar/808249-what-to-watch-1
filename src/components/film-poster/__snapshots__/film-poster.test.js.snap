// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Poster of the film correctly renders when isBig not passed div render with class 'movie-card__poster' 1`] = `
<div
  className="movie-card__poster "
>
  <img
    alt="Avengers: Endgame poster"
    height="327"
    width="218"
  />
</div>
`;

exports[`Poster of the film correctly renders when isBig passed div render with classes 'movie-card__poster' and 'movie-card__poster--big' 1`] = `
<div
  className="movie-card__poster movie-card__poster--big"
>
  <img
    alt="Avengers: Endgame poster"
    height="327"
    width="218"
  />
</div>
`;
